### ISSUE TITLE
listresources fails in omni2

### ISSUE THREAD

**Issue by ahelsing** [GENI-NSF/geni-portal#238]
In omni 2.0 for AM API v2+, the return is keyed by AM URL, not the URN/URL pair.

Also, the value returned is the full code/value/output struct, not just the rspec

listresources.php and print-text-helpers.php both need updates to handle this.

 _Imported from trac ticket [#238](http://trac.gpolab.bbn.com/proto-ch/ticket/238),  created by ahelsing on 09-24-2012 at 14:11, last modified: 09-24-2012 at 15:55_

**Comment by ahelsing** [comment]
Errors from Marshall:
```

I allocated a PC on pgeni3 and then asked for the manifest and got these errors in the log:

[Mon Sep 24 13:47:50 2012] [error] [client 128.89.81.89] am_client invoke_omni_function COMMAND = /usr/share/geni-ch/portal/gcf/src/omni_php.py -c /tmp/mbrinn-omni.ini -l /usr/share/geni-ch/portal/gcf/src/logging.conf --logoutput /tmp/omni.log --api-version 2 --GetVersionCacheName /tmp/omniVersionCacheplEeqS --slicecredfile /tmp/mbrinn.slicecredential -t GENI 3 listresources urn:publicid:IDN+marilac:AAA+slice+BBB, referer: https://marilac.gpolab.bbn.com/secure/slice.php?project_id=886e4f0a-7d3f-4a02-8595-18d726cd4c79&slice_id=8e83a200-d0dd-4c90-8b40-bc587fb4ad3e&member_id=
[Mon Sep 24 13:47:54 2012] [error] [client 128.89.81.89] ListResources output = Array, referer: https://marilac.gpolab.bbn.com/secure/slice.php?project_id=886e4f0a-7d3f-4a02-8595-18d726cd4c79&slice_id=8e83a200-d0dd-4c90-8b40-bc587fb4ad3e&member_id=
[Mon Sep 24 13:47:55 2012] [error] [client 128.89.81.89] PHP Notice:  Undefined offset: 0 in /var/www/secure/print-text-helpers.php on line 290, referer: https://marilac.gpolab.bbn.com/secure/slice.php?project_id=886e4f0a-7d3f-4a02-8595-18d726cd4c79&slice_id=8e83a200-d0dd-4c90-8b40-bc587fb4ad3e&member_id=
[Mon Sep 24 13:47:55 2012] [error] [client 128.89.81.89] PHP Notice:  Undefined offset: 1 in /var/www/secure/print-text-helpers.php on line 291, referer: https://marilac.gpolab.bbn.com/secure/slice.php?project_id=886e4f0a-7d3f-4a02-8595-18d726cd4c79&slice_id=8e83a200-d0dd-4c90-8b40-bc587fb4ad3e&member_id=
[Mon Sep 24 13:47:55 2012] [error] [client 128.89.81.89] PHP Notice:  Undefined index:  in /var/www/secure/am_map.php on line 48, referer: https://marilac.gpolab.bbn.com/secure/slice.php?project_id=886e4f0a-7d3f-4a02-8595-18d726cd4c79&slice_id=8e83a200-d0dd-4c90-8b40-bc587fb4ad3e&member_id=
[Mon Sep 24 13:47:55 2012] [error] [client 128.89.81.89] Call to print_rspec_pretty() FAILED, referer: https://marilac.gpolab.bbn.com/secure/slice.php?project_id=886e4f0a-7d3f-4a02-8595-18d726cd4c79&slice_id=8e83a200-d0dd-4c90-8b40-bc587fb4ad3e&member_id=
[Mon Sep 24 13:47:55 2012] [error] [client 128.89.81.89] PHP Notice:  Undefined variable: am_id in /var/www/secure/listresources.php on line 137, referer: https://marilac.gpolab.bbn.com/secure/slice.php?project_id=886e4f0a-7d3f-4a02-8595-18d726cd4c79&slice_id=8e83a200-d0dd-4c90-8b40-bc587fb4ad3e&member_id=
```


_Trac comment by ahelsing on 09-24-2012 at 14:12_

**Comment by seledwards** [comment]
Ok.  I fixed `print-text-helpers.php` to handle the 2.0 output.

While I was in there, I suppressed output from AMs which can't find the slice (presumably because you haven't reserved resources there).  I only suppress it for the pretty version of the output, in the "Raw RSpec" view, it'll print an error message returned by the AM instead.

Closing.

_Trac comment by sedwards (github user: seledwards) on 09-24-2012 at 15:55_
